### DISCUSSION TITLE
Track new polygons for boolean operations

### DISCUSSION THREAD

**Discussion by ErikSom** [jscad/OpenJSCAD.org#1170] [Q&A]
Hi,

I'm using boolean operations on my Mesh and I was wondering if there is a reliable way to figure out which polygons are newly created and which polygons are from the pre-operation mesh.

Thanks in advance!

**Comment by z3dev** [comment]
Really good question... I believe that original untouched polygons are returned untouched. You might want to try this of course. If that's the case then you could add an id to each polygon, and check later.

**Reply by ErikSom** [reply]
It seems all polygons are renewed after the operation

**Comment by hrgdavor** [comment]
@z3dev we have discussed this further in discord. And I understood that @ErikSom is actually looking for the option that jsacd used to have... in older versions jscad would keep polygon colors if you boolean shapes with different color. I remember mentions that it was removed for performance...
